### ISSUE TITLE
Update maps 

### ISSUE THREAD

**Issue by farruxx** [mapsme/omim#8773]
How can I update maps?
I researched this repo and couldn't understand if there is a way to update maps without updating apk file.
Also in android java files there is `UpdaterDialogFragment`, how can I trigger it when new maps are available on server?

**Comment by thany** [comment]
You could overwite your maps manually with these:
http://direct.mapswithme.com/direct/latest/

Not sure how well they will work with older versions. Usually the version of map files is baked into the APK. In other words, an update to the APK will almost always trigger downloading map updates.

**Comment by tuxayo** [comment]
latest/ isn't up to date anymore http://direct.mapswithme.com/direct/

**Comment by thany** [comment]
HTTP 403. Looks like they're really forcing folks to update the app. Nasty move.
I wonder if someone can try to figure out which url's the app itself is using to update now. I don't have the time or know-how to do that - I wish I did.

**Comment by tatiana-yan** [comment]
Actually application is opensource but maps from our server are not allowed to use outside of official application. It's illegal.
You can generate own maps using generator_tool.

**Comment by thany** [comment]
Illegal?! Under which law?

Is there any proper documentation on how to install and use that tool? Or is it one of those "works on my machine" tools that requires 23981 steps for every square millimetre of mapping? Cuz I've seen this happening.

**Comment by tatiana-yan** [comment]
Illegal under maps.me licence: you can use our code, it's open. But all images and media content including maps from our server are under copyright, see

https://github.com/mapsme/omim/blob/master/NOTICE

**Comment by tatiana-yan** [comment]
https://github.com/mapsme/omim/blob/master/docs/MAPS.md

**Comment by thany** [comment]
Okay, so not illegal at all.
All you have to do is give permission and we're done. Why wouldn't you do that?

**Comment by tuxayo** [comment]
Any contact or terms of use have legal value from what I understand.
Of course if they contradict the law of a country for example, they would be void/illegal in that country.

**Comment by tuxayo** [comment]
Shouldn't the map data of maps.me be libre?
Due to the OpenStreetMap data license? (ODbL)
https://en.wikipedia.org/wiki/Open_Database_License

**Comment by tuxayo** [comment]
@thany 

> I wonder if someone can try to figure out which url's the app itself is using to update now. I don't have the time or know-how to do that - I wish I did.

Indeed, it seems that this very discussion has caused directory listing to be disabled.
Archived versions of the directory:
https://web.archive.org/web/20190112111508/http://direct.mapswithme.com/direct/
https://archive.is/SRuMS

@tatiana-yan Sorry to cause trouble. Is that a treat to your business model?
If so, can we imagine solutions like recurring crowdfunding? Where if an amount is reached, МАРS.ME won't need to do things like closing the assets.

Possible platforms:
- https://liberapay.com/ , there is a pledge system which [some people have used to pledge giving to MAPS.ME](https://liberapay.com/on/github/mapsme) :) 
- https://opencollective.com/
- https://www.bountysource.com/
- https://www.patreon.com/ (non-libre)

There must be an amount, everyone has a price ;)
I would be happy to give a few bucks as I'm using and recommending MAPS.ME a lot.

Even if you think that won't work, leaving the possibility will show good faith towards allowing better funding models of libre/open source software.
And will also allow people that use the app but avoid the PlayStore (such as me) to give money to you. Because your app is the most user friendly way to use OSM and deserves funding.
Thanks for your time considering this.

**Comment by thany** [comment]
Closing up access to data is not exactly generating goodwill, especially since you got the data for free in the first place. Hell, it is even sponsored with booking.com data. Many governments or other local authorities pour their map data into OSM, totally for free and out of nothing but good intentions.

You're grabbing all that juicy data, packaging it up, and then basically DRM'ing it. Very, very uncool.

**Comment by tuxayo** [comment]
@thany 
> Closing up access to data is not exactly generating goodwill
> [...]·
> You're grabbing all that juicy data, packaging it up, and then basically DRM'ing it. Very, very uncool.

I'm hopping to convince them that it's not the best way. It's really great not being alone from the community in this discussion. Do you have ideas? Something that can help having a happier discussion towards opening the data? We didn't yet had the occasion to hear more from them. And it's important that we get as much dialogue as possible. I hope that how the chances to get to a happy conclusion are the bests.

In many cases keeping as much exclusivity on intellectual property as possible in the expected default choice. That's our current world, but that can change. And that's why it's important to spread the ideas of libre/open culture (because that's not obvious) and to try to understand the funding models to see if that's really necessary and suggest [1] models that allow more openness.
Because in the end, economy/funding [2] is the key for good and bad.

[1] and be willing to pay, because there is no chance otherwise, like when going against advertisement
[2] in the broad sense, the way (more of less freely) that some people can put time on some work

**Comment by burivuh** [comment]
Our maps are enriched by proprietary data (for example, Booking.Com did not grant us rights to open their data provided by a contract).
This is the biggest issue.

Then, we see no point in providing and supporting anybody with the resources spent on map building, testing and map distribution. If we will have a valid request from HOT OSM or similar organisations we may provide them with some custom maps, others can build maps by themselves using our open sourced data generator. These maps will be based only on the OSM.

**Comment by thany** [comment]
> Our maps are enriched by proprietary data (for example, Booking.Com did not grant us rights to open their data provided by a contract).
> This is the biggest issue.

That is a problem you took upon yourselves. Also booking.com themselves draw from data provided to them by third parties (or fourth parties I guess) so in no way should they have the right(1) to make such demands.

> Then, we see no point in providing and supporting anybody with the resources spent on map building, testing and distribution. If we will have a valid request from HOT OSM or similar organisations we may provide them with some custom maps, others can build maps by themselves using our open sourced data generator. These maps will be based only on the OSM.

You almost provided your own solution. Provide the maps freely without proprietary data embdedded. Provide the proprietary data in separate files to the official app(2). Then you're being honest to the community, and are distributing OSM data in an ethical way.


(1) I don't mean legal right, but more like ethical right.
(2) With all the DRM and other copy protection crap you feel neccesary.

**Comment by burivuh** [comment]
1. Well, it is the way it is. 
2. We already provided everybody with the solution. You can freely create your own based on it.

**Comment by h1z1** [comment]
Interesting thread.  OSM data is used by other commercial GIS products I gurantee you can't afford their licenses for.  Pretty arrogant replies from people assuming providing those maps is somehow free too.  Out of Russia of all places.     I've been with people in the middle of nowhere without data, it's a damn good feeling not being tied to Google.   For that reason if anything some easier way to share between devices would be helpful - torrent. bluetooth, etc.

> All you have to do is give permission and we're done. Why wouldn't you do that?

There's nothing in the ODbL that requires data be provided in a specific format.  Even the DRM clause.

**Comment by thany** [comment]
Not sure about the others, but I'm not Russian. I'm from the EU, where open data is becoming pretty normal. Bare in mind that Maps.me is essentially opensource, which may change the terms under which data can be provided. Commercial software may fall under different conditions.

Someone should probably dive into the OSM license and see if their data is allowed to be repackaged as-is and sold for a pile of money.

**Comment by h1z1** [comment]
What do you think "Opendata" is?  This has nothing to do with it, you're effectively supporting the very exploitation you claim exists in this thread.

**Comment by thany** [comment]
It has nothing to do with it. I merely took opendata as an example that opening up previously proprietary data to the general public is gaining traction in the EU.
